pkg/resource/deploy/source_eval.go

```diff
@@ -1010,6 +1010,8 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 				return nil, fmt.Errorf("%s: passed invalid version: %w", label, err)
 			}
 			providers.SetProviderVersion(props, &version)
+		} else if prov, ok := rm.defaultProviders.defaultProviderInfo[providers.GetProviderPackage(t)]; ok && prov.Version != nil {
```

Comment:
Can you have downloadURL set but version not set? If somehow the user tries to use a plugin without a version but with a download url won't this line then cause that plugin to not be found?